### PR TITLE
Task 94564: add way to access httpserver (AB#94564)

### PR DESCRIPTION
**Pull request by Limpaan**

### CODE REVIEW

File: authentication.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
+const jwt = require('jsonwebtoken')
+const { jwksOptions, validationOptions } = require('./authenticationOptions')
+const { JwksClient } = require('jwks-rsa')
+
+async function authenticate (authorization) {
```
Comment (ohritz) [inline authentication.js:5]: Any reason why we wouldnt use https://github.com/auth0/node-oauth2-jwt-bearer/tree/main/packages/express-oauth2-jwt-bearer ?

File: index.js [hunk 2]
```diff
@@ -26,36 +28,61 @@ const expressNemo = options => {
...
+      .get('/health', middlewares.health)
+
+    return {
+      nemoApp: nemoApp,
+      routedApp: routedApp
+    }
+  }
+
+  const startApp = (listenServer) => {
+    const server = listenServer.listen(PORT, () =>
```
Comment (esepato) [inline index.js:47]: Parameter should be named app?

Comment (Limpaan) [reply index.js]: ye, makes sense

File: authentication.js (hunk 1 above)
Comment (Limpaan) [reply authentication.js:5]: https://github.com/auth0/node-oauth2-jwt-bearer/tree/main/packages/access-token-jwt this is what we could use, but that's not published yet so I think it'd be unwise to rely on it

Comment (ohritz) [reply authentication.js:5]: My bad.. Carry on...

File: authentication.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+  try {
+    jwt.verify(token, publicKey, validationOptions)
+  } catch (err) {
+    throw new Error('Could not verify jwt, check if token is correct')
+  }
+
+  return {
+    user: decodedToken.payload
+  }
```
Comment (ohritz) [inline authentication.js:48]: seems wrong to call the jwt token payload user. It could potentially be anything.

**Review by ohritz — APPROVED** [review]

**Pull request merged**